server/sonar-db-dao/src/main/java/org/sonar/db/ce/CeQueueDao.java

```diff
@@ -84,6 +85,10 @@ public int countByQuery(DbSession dbSession, CeTaskQuery query) {
     return mapper(dbSession).selectPendingByMinimumExecutionCount(minExecutionCount);
   }
 
+  public void resetTasksWithUnknownWorkerUUIDs(DbSession dbSession, Set<String> knownWorkerUUIDs) {
+    mapper(dbSession).resetTasksWithUnknownWorkerUUIDs(knownWorkerUUIDs, system2.now());
```

Comment:
forget my last paragraph. I though you meant to modify the DAO parameter. The Mapper method parameter indeed has to be changed to a List.